src/components/ssc-exercise/index.tsx

```diff
@@ -95,54 +99,73 @@ const SSCExercise: React.FC<SSCExerciseProps> = ({chapter, steps}) => {
   };
 
   return (
-    <PageLayout
-      backgroundColor={sscMock[chapter as Exclude<keyof typeof sscMock, 'about'>].backgroundColor}
-      header={
-        <Header onBackButton={previousStep}>
-          <CarouselPagination steps={steps} currentStep={currentStep} />
-        </Header>
-      }
-      footer={
-        currentStep === steps.length - 1 ? (
-          <Button variant="blue" className="mx-auto" onClick={handleComplete}>
-            {t('completeButton')} <Complete />
-          </Button>
-        ) : (
-          <Button variant={getButtonVariant} onClick={nextStep}>
-            {t('nextStep')} <ArrowRight />
-          </Button>
-        )
-      }>
-      <section className="flex h-full w-full flex-1 items-center justify-center">
-        <Carousel className="h-full w-full flex-1" setApi={setApi}>
-          <CarouselContent>
-            {steps.map((_, index) => (
-              <CarouselItem key={index} className="space-y-6">
-                <h1 className="text-2xl font-bold">{steps[index].title}</h1>
-                {splitTextIntoParagraphs(steps[index].description).map((paragraph, index) => (
-                  <div key={index} className="flex items-center">
-                    <div>
-                      <div className="bg-black w-2 h-2 rounded-full mx-2"></div>
+    <>
+      {isDialogOpen ? (
+        <DialogView
```

Comment:
The dialog is not supposed to be rendered condiitionally. There are props like "open" and "onOpenChange" on the dialog that can consume your dialog hook. But I'll approve this now and we can improve it later

Reply:
https://www.radix-ui.com/primitives/docs/components/dialog